### PR TITLE
Split time-tests common library

### PR DESCRIPTION
**Pull request by asomsiko**
Applied clang-format-9 and added a README.md

### CODE REVIEW

File: tests/time_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+This test suite contains pipelines, which are executables. The
+pipelines measure the time of their execution, both total and 
+partial. A Python runner calls the pipelines and calcuates the 
+average execution time.
+
+## Prerequisites
+
+To build the time tests, you need to have the `build` folder, 
+which is created when you configure OpenVINO.
```
Comment (vurusovs) [inline tests/time_tests/README.md:11]: should we add a link for `OpenVINO how to build` docs?

File: tests/time_tests/src/infer-pipeline.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2020 Intel Corporation
```
Comment (vurusovs) [inline tests/time_tests/src/infer-pipeline.cpp:1]: @asomsiko could you resolve issue with added/removed files to see a Git difference between corresponding files? At the moment `ftti_pipeline.h` marked as removed, and `infer-pipeline.cpp` as newly added

File: tests/time_tests/time-testhelper/main.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "cli.h"
+#include "statistics_writer.h"
```
Comment (vurusovs) [inline tests/time_tests/time-testhelper/main.cpp:6]: @asomsiko should we move the file to `time-testhelper` folder?

File: tests/time_tests/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Time Tests
+
+This test suite contains pipelines, which are executables. The
+pipelines measure the time of their execution, both total and 
+partial. A Python runner calls the pipelines and calcuates the 
+average execution time.
```
Comment (vurusovs) [inline tests/time_tests/README.md:6]: @asomsiko I propose to change `average` to `aggregated` to give us more freedom in metrics changes

File: tests/time_tests/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+To build and run the tests, open a terminal and run the commands
+below:
+1. Build tests:
+``` bash
+cmake .. -DInferenceEngineDeveloperPackage_DIR=../../../build && make time-tests
+```
+2. Run test:
+``` bash
+../../../bin/intel64/Release/infer-pipeline -m model.xml -d CPU
```
Comment (vurusovs) [inline tests/time_tests/README.md:23]: this code may be not used as is in case of `Debug` version

File: tests/time_tests/src/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# add dummy `time_tests` target combines all time tests
+add_custom_target(time_tests)
+
+# Build test from every source file matchs *-pipeline.cpp.
+# Test target name is source file name without extension.
```
Comment (vurusovs) [inline tests/time_tests/src/CMakeLists.txt:9]: @asomsiko should we extend target name with some unique prefix to determine belonging to `time-tests`? At the moment `infer-pipeline` isn't clear for an outside developer

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: tests/time_tests/time-testhelper/main.cpp (hunk 3 above)
Comment (asomsiko) [reply tests/time_tests/time-testhelper/main.cpp:6]: No, since that is not a public API of time-testhelper

File: tests/time_tests/README.md (hunk 4 above)
Comment (asomsiko) [reply tests/time_tests/README.md:6]: The Readme is for current state. The runner does display average. It also does many other things which we may explain another PR.

File: tests/time_tests/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+```
+
```
Comment (vurusovs) [inline tests/time_tests/README.md:25]: no info about `run_executable.py`

File: tests/time_tests/README.md (hunk 5 above)
Comment (vurusovs) [reply tests/time_tests/README.md:23]: Decided to leave it as is because it's default folder layout

**Review by aalborov — APPROVED** [review]

File: tests/time_tests/src/timetest_infer.cpp [hunk 8]
```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <inference_engine.hpp>
+#include <iostream>
+
+#include "time-testhelper/timer.h"
```
Comment (vurusovs) [inline tests/time_tests/src/timetest_infer.cpp:8]: @asomsiko will you update `time-testhelper` name?

**Review by vurusovs — APPROVED** [review]

File: tests/time_tests/src/timetest_infer.cpp (hunk 8 above)
Comment (asomsiko) [reply tests/time_tests/src/timetest_infer.cpp:8]: Will need separate PR for this

**Pull request merged**